src/DistributedLock.Postgres/PostgresDistributedReaderWriterLock.cs

```diff
@@ -29,6 +30,14 @@ public PostgresDistributedReaderWriterLock(PostgresAdvisoryLockKey key, IDbConne
     {
     }
 
+    /// <summary>
+    /// Constructs a lock with the given <paramref name="key"/> (effectively the lock name) and <paramref name="npgsqlDataSource"/>.
+    /// </summary>
+    public PostgresDistributedReaderWriterLock(PostgresAdvisoryLockKey key, NpgsqlDataSource npgsqlDataSource)
```

Comment:
Let's have this take `DbDataSource` instead of `NpgsqlDataSource`. That allows people who wrap the underlying provider in their own abstraction to still use it.

Reply:
Swapped to DbDataSource instead!